pallets/nft/src/tests.rs

```diff
@@ -3,159 +3,732 @@ use crate::mock::{Event, *};
+		assert_noop!(
+			Nft::transfer(Origin::signed(bob_account.clone()), alice_account.clone(), (1, 10),),
+			// If transfer an non-exist token id, the error will be from orml_nft, who's Event or Error is not defined
+			orml_nft::Error::<Test>::TokenNotFound
+		);
+
+		// someone else transfer no-owned token
+		assert_noop!(
+			Nft::transfer(Origin::signed(alice_account.clone()), bob_account.clone(), (1, 0),),
+			// If transfer an no-permission token id, the error will be from orml_nft, who's Event or Error is not defined
+			orml_nft::Error::<Test>::NoPermission
+		);
+
+		// normal transfer
+		assert_ok!(Nft::transfer(
+			Origin::signed(bob_account.clone()),
+			alice_account.clone(),
+			(1, 0),
+		));
+		assert_eq!(
+			events_filter::<crate::Event::<Test>>()[4],
+			Event::Nft(crate::Event::TransferredToken(
+				bob_account.clone(),
+				alice_account.clone(),
+				1,
+				0
+			)),
+		);
+	});
+}
+
+#[test]
+// Test for Simple type Event:
+// MintedToken(T::AccountId, T::AccountId, ClassIdOf<T>, u32)
+
+// Test for Error:
+// ClassIdNotFound
+// NoPermission,
+// InvalidQuantity,
+// QuantityOverflow
+// OutOfCampaignPeriod
+
+fn test_minted_token_event() {
+	new_test_ext().execute_with(|| {
+		let (alice_account, bob_account) = initial_account();
+		// create Transferable Unburnable class without start/end restrcition
+		assert_ok!(Nft::create_class(
+			Origin::signed(alice_account.clone()),
+			CID::default(),
+			Properties(ClassProperty::Transferable.into()),
+			None,
+			None,
+			ClassType::Simple(10),
+		));
+
+		//mint class Id=1 non-exist class
+		assert_noop!(
+			Nft::mint(
+				Origin::signed(alice_account.clone()),
+				bob_account.clone(),
+				1,
+				CID::default(),
+				1,
+			),
+			NftError::ClassIdNotFound
+		);
+
+		//mint 0 invalid instance quantity
+		assert_noop!(
+			Nft::mint(
+				Origin::signed(alice_account.clone()),
+				bob_account.clone(),
+				0,
+				CID::default(),
+				0,
+			),
+			NftError::InvalidQuantity
+		);
+
+		//mint 11 exceed the maximum instance limit
+		assert_noop!(
+			Nft::mint(
+				Origin::signed(alice_account.clone()),
+				bob_account.clone(),
+				0,
+				CID::default(),
+				11,
+			),
+			NftError::QuantityOverflow
+		);
+
+		//mint 5 instance with right ClassInfo owner
+		assert_ok!(Nft::mint(
+			Origin::signed(alice_account.clone()),
+			bob_account.clone(),
+			0,
+			CID::default(),
+			5,
+		));
+		assert_eq!(
+			events_filter::<crate::Event::<Test>>()[1],
+			Event::Nft(crate::Event::MintedToken(alice_account.clone(), bob_account.clone(), 0, 5))
+		);
 
-		// root is 0xa8a5ec29a3df3c5a8aa6fd2935d2414cf0ce4f748a13bb2833214c3b94a6d3b3
+		//mint 5 instance with wrong ClassInfo owner
+		assert_noop!(
+			Nft::mint(
+				Origin::signed(bob_account.clone()),
+				bob_account.clone(),
+				0,
+				CID::default(),
+				5,
+			),
+			NftError::NoPermission
+		);
+
+		// create Transferable Unburnable class with start/end restrcition
```

Comment:
I would think these kind of test be in another test function only to test start/end restrcition for all the cases. You may have a issue for this instead of fix it now

Reply:
Will have an issue for this.